reindexer/reindex_worker/src/main/scala/uk/ac/wellcome/platform/reindex/reindex_worker/services/ReindexWorker.scala

```diff
@@ -1,31 +1,109 @@
 import com.google.inject.Inject
+import com.gu.scanamo.{DynamoFormat, ScanamoFree}
+import com.twitter.inject.annotations.Flag
+import io.circe.Encoder
 import uk.ac.wellcome.messaging.sns.NotificationMessage
 import uk.ac.wellcome.messaging.sqs.SQSStream
 import uk.ac.wellcome.platform.reindex.reindex_worker.models.ReindexJob
 import uk.ac.wellcome.json.JsonUtil._
-import uk.ac.wellcome.storage.vhs.HybridRecord
+import uk.ac.wellcome.platform.reindex.reindex_worker.exceptions.ReindexerException
+import uk.ac.wellcome.storage.dynamo._
+import uk.ac.wellcome.storage.vhs.{EmptyMetadata, HybridRecord, VHSIndexEntry}
 
 import scala.concurrent.{ExecutionContext, Future}
 
+case class MiroMetadata(showInCatalogueAPI: Boolean)
+
 class ReindexWorker @Inject()(
   recordReader: RecordReader,
-  hybridRecordSender: HybridRecordSender,
+  bulkSNSWriter: BulkSNSWriter,
   system: ActorSystem,
-  sqsStream: SQSStream[NotificationMessage]
+  sqsStream: SQSStream[NotificationMessage],
+  @Flag("reindexer.tableMetadata") tableMetadata: String
 )(implicit ec: ExecutionContext) {
   sqsStream.foreach(this.getClass.getSimpleName, processMessage)
 
+  implicit val emptyMetadataDynamoFormat: DynamoFormat[EmptyMetadata] =
+    DynamoFormat[EmptyMetadata]
+
+  implicit val emptyMetadataEncoder: Encoder[EmptyMetadata] =
+    Encoder[EmptyMetadata]
+  implicit val miroMetadataEncoder: Encoder[MiroMetadata] =
+    Encoder[MiroMetadata]
+
   private def processMessage(message: NotificationMessage): Future[Unit] =
     for {
       reindexJob: ReindexJob <- Future.fromTry(
         fromJson[ReindexJob](message.body))
-      outdatedRecords: List[HybridRecord] <- recordReader
-        .findRecordsForReindexing(reindexJob)
-      _ <- hybridRecordSender.sendToSNS(records = outdatedRecords)
+      items <- recordReader.findRecordsForReindexing(reindexJob)
+      _ <- tableMetadata match {
+        case "MiroMetadata" => processMiroMetadata(items)
+        case _              => processEmptyMetadata(items)
+      }
     } yield ()
 
-  def stop() = system.terminate()
+  def stop(): Future[Terminated] = system.terminate()
+
+  /** These are the handlers that take the Dynamo rows, convert them into case
+    * classes, and send them to SNS.
+    *
+    * We can't commonise this code any further, because we send different types
+    * depending on whether the row has any metadata.
+    *
+    * In particular, we send a `HybridRecord` if the row doesn't have any
+    * metadata, and `VHSIndexEntry` otherwise.
+    */
+  private def processEmptyMetadata(
+    items: List[util.Map[String, AttributeValue]]): Future[Unit] =
+    bulkSNSWriter
+      .sendToSNS(items.map { av =>
+        // AWLC: if you're changing this code to create an instance of
+        // `EmptyMetadata` to include in a `VHSIndexEntry`, don't bother
+        // parsing as a case class; just create a fresh value.
+        //
+        // When I tried to call `parseAsCaseClass[EmptyMetadata]`, I got
```

Comment:
Note to self: this turned out to be caused by a Circe problem (specifically, a null decoder was being passed). Revisit this?